### PR TITLE
PRO-3692-Warning Messages

### PR DESCRIPTION
**Pull request by vignesha22**
## Description

- Updated Warning Messages
- Fixed alignment of warning sign
- Add native balance check on the max USDC token available from balance on buy
- Changed minimum buy amount to 2 USD

## How Has This Been Tested?

- Tested Locally

## Screenshots (if appropriate):
Screenshot 2025-10-07 at 7 11 07 PM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary by CodeRabbit

* **New Features**
  * Aggregates stable balances per chain; uses total for Buy flow and selects chain with largest stable balance for gas checks; shows native-token and chain details in messages.
  * Wallet balance display and MAX button use aggregated stable balance; minimum buy amount raised to $2.

* **Bug Fixes**
  * Buy button now disables immediately when liquidity is insufficient; clearer low-balance and gas-fee warnings.

* **Tests**
  * Expanded tests and mocks for new warnings, edge cases, dispensable-assets, and updated wallet portfolio shape.

### CODE REVIEW

File: src/apps/pulse/components/Buy/Buy.tsx [hunk 1]
```diff
@@ -217,6 +267,31 @@ export default function Buy(props: BuyProps) {
...
+    // Take the Max value of stable currency balance across chains and also take the chainId
+    const maxStableBalance = Math.max(...Object.values(stable_Balance));
+    const chainIdOfMaxStableBalance = Number(
+      Object.keys(stable_Balance).find(
+        (key) => stable_Balance[Number(key)] === maxStableBalance
+      ) || '1'
+    );
+    if (!walletPortfolioData) return;
+    // Take the native token balance of that chain
+    const tokens = convertPortfolioAPIResponseToToken(walletPortfolioData.result.data);
```
Comment (RanaBug) [inline src/apps/pulse/components/Buy/Buy.tsx:281]: I feel tokens, and nativeToken might be already showing or coded somewhere in the Pulse app, to avoid repetition here

File: src/apps/pulse/components/Buy/Buy.tsx [hunk 2]
```diff
@@ -583,11 +658,15 @@ export default function Buy(props: BuyProps) {
...
-                message = 'Min. amount 0.5 USD';
+                message = 'Min. amount 2 USD';
               } else if (insufficientWalletBalance) {
                 message = 'Insufficient wallet balance';
               } else if (notEnoughLiquidity && token) {
                 message = 'Not enough liquidity';
+              } else if (minimumStableBalance) {
+                message = 'You need $2 USDC to trade, deposit USDC';
+              } else if (minGasFee) {
+                message = `You need at least $1 of ${minGasFeeChain} to trade, deposit ${minGasFeeChain}`;
```
[the remark this answers is not in the document]
Comment (vignesha22) [reply src/apps/pulse/components/Buy/Buy.tsx:670]: it automatically gets the symbol of that chainId and store it in minGasFeeChain which is then displayed to the user

File: src/apps/pulse/utils/blockchain.ts [hunk 3]
```diff
@@ -7,3 +7,25 @@ export const formatNativeTokenAddress = (address: string): string => {
...
+  137: 'POL',
+  8453: 'ETH',
+  42161: 'ETH',
+  10: 'ETH',
+  56: 'BNB',
+};
+
+export const getNativeTokenSymbol = (chainId: number): string => {
+  return NativeSymbols[chainId] || 'ETH';
+};
```
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pulse/utils/blockchain.ts:31]: @vignesha22 please review this. Do you have prettier?

File: src/apps/pulse/components/Buy/Buy.tsx [hunk 4]
```diff
@@ -198,6 +243,11 @@ export default function Buy(props: BuyProps) {
           setNoEnoughLiquidity(true);
           return;
         }
+        const stableBalance = getStableCurrencyBalance();
```
[the remark this answers is not in the document]
Comment (vignesha22) [reply src/apps/pulse/components/Buy/Buy.tsx:246]: no it stores the USD value and for any chain the USD value is the same and that's why you are able to get total stable coin balance as one and show it to the user in that please see the bottom of input USD that's the total value of all stable coins

File: src/apps/pulse/components/Buy/Buy.tsx [hunk 5]
```diff
@@ -150,6 +154,47 @@ export default function Buy(props: BuyProps) {
     );
   };
 
+  const getStableCurrencyBalanceOnEachChain = () => {
```
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pulse/components/Buy/Buy.tsx:157]: what is the difference between getStableCurrencyBalance and getStableCurrencyBalanceOnEachChain? And Why do we need both?

File: src/apps/pulse/components/Buy/Buy.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pulse/components/Buy/Buy.tsx:246]: I mean for pulse and intentSDK, currently it handles USD one by one not together. So in my example above, you cannot buy tokens for 2 USD, but for a max of 1.7 USD (the highest single USD balance)

File: src/apps/pulse/components/Buy/Buy.tsx [hunk 6]
```diff
@@ -198,6 +219,12 @@ export default function Buy(props: BuyProps) {
           setNoEnoughLiquidity(true);
           return;
         }
+
+        if (sumOfStableBalance < 2) {
```
Comment (RanaBug) [inline src/apps/pulse/components/Buy/Buy.tsx:223]: We are still looking here at the total sum of stable balance. However Pulse at the moment is not using the total some of USD to buy a token, it looks at all balances, and chose the balance of the USD that could be enough to buy the tokem

File: src/apps/pulse/components/Buy/Buy.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (aldin4u) [reply src/apps/pulse/components/Buy/Buy.tsx]: @vignesha22  in the scenario that user is inputting within the cumulative balance of USDC across all chains. User inputs a balance that is not part of a single chain. Lets say cumalative balance is $20 but $10 is on Optimism, $5 is on Arbitrum and $5 is on Base. The warning message should state: "Not enough USDC, reduce amount to $10. "
